### PR TITLE
Remove vestiges of Python 2

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

This pull request removes the few remaining traces of Python 2, except the handful from `astropy/extern`.

The `except`-block from `astropy/utils/data.py` has been removed under the assumption that it is not needed anymore, but because the problem it was meant to fix was intermittent it might be required after all. The removal of that `except`-block is therefore in a separate commit so that it could be easily reverted should the need arise.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Failures might be related?

**Review by mhvk — APPROVED** [review]
Thanks, @eerovaher, that looks all great -- I have no comments whatsoever (though we need to get the tests working again...)

**Comment by mhvk** [comment]
See #12682 for an attempt to get the tests OK again.

File: astropy/utils/data.py [hunk 1]
```diff
@@ -1379,13 +1378,6 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None,
                                      + remote_url)
                 e.reason.args = (e.reason.errno, e.reason.strerror)
             errors[source_url] = e
-        except socket.timeout as e:
```
Comment (pllim) [inline astropy/utils/data.py:1382]: I am not 100% sure this is not a problem in Python 3 also. Can anyone confirm?

File: astropy/table/operations.py [hunk 2]
```diff
@@ -951,7 +951,9 @@ def get_descrs(arrays, col_name_map):
             raise TableMergeError(f'Key columns {names!r} have different shape')
         shape = uniq_shapes.pop()
 
-        out_descrs.append((fix_column_name(out_name), dtype, shape))
+        if out_name is not None:
```
Comment (pllim) [inline astropy/table/operations.py:954]: I see you use in-line `if` to do such conversion, why not make that style consistent throughout this PR?

**Comment by eerovaher** [comment]
I've expanded this pull request slightly so that it removes `__div__()` from everywhere, not just from `astropy/coordinates/representation.py`.

File: astropy/table/operations.py (hunk 2 above)
Comment (eerovaher) [reply astropy/table/operations.py:954]: I used the one-liner if the variable on the left hand side had a different name than the variable on the right hand side, but if you think it's worth it then I could use it here too.

**Review by mhvk — APPROVED** [review]
To me, this looks OK. @pllim is right in worrying about the socket timeout error, but since it was only occasional, my sense would be to try merging this (i.e., hope that we remember the problem if we get occasional failures...).

File: astropy/table/operations.py (hunk 2 above)
Comment (pllim) [reply astropy/table/operations.py:954]: Naw, it's ok. Was just a nitpick.

**Review by pllim — APPROVED** [review]
Okay then. As per @mhvk 's comment, let's get this in but we need to revert the socket stuff if the error comes back.

**Comment by pllim** [comment]
Thanks for the nice cleanup!

**Pull request merged**